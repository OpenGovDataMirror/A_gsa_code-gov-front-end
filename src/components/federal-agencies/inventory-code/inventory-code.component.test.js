import React from 'react'
import { shallow } from 'enzyme'
import InventoryCodeComponent from './inventory-code.component'

const props = {
  schema: {},
  url: 'http://test.localdomain/assets/data/schema.json'
}

let wrapper
describe('components - InventoryCodeComponent', () => {
  beforeEach(() => {
    wrapper = shallow(<InventoryCodeComponent {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render optional entries on load', () => {
    expect(wrapper.find('#schema-viewer').prop('className')).not.toContain('hide-optional-fields')
  })

  it('should render toggle optional entries on click', () => {
    wrapper.find('#hide-optional-fields').simulate('click')
    expect(wrapper.state('optionalFields')).toBeTruthy()
    wrapper.find('#hide-optional-fields').simulate('click')
    expect(wrapper.state('optionalFields')).toBeFalsy()
  })
})
