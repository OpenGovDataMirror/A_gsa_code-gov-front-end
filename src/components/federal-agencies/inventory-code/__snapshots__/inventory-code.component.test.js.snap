// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - InventoryCodeComponent should render correctly 1`] = `
<div
  className="margin-top-4 "
  id="schema-viewer"
>
  <form
    className="usa-form"
    id="optional-toggle-form"
  >
    <div
      className="usa-checkbox"
    >
      <input
        className="usa-checkbox__input"
        id="hide-optional-fields"
        onClick={[Function]}
        type="checkbox"
        value="hide-optional-fields"
      />
      <label
        className="usa-checkbox__label"
        htmlFor="hide-optional-fields"
      >
        Hide Optional Fields
      </label>
    </div>
  </form>
  <div
    className="desktop-and-mobile-views"
  >
    <InventoryCodeTableComponent />
  </div>
</div>
`;
